src/s3/commands/presignedURL.ts

```diff
@@ -41,33 +47,115 @@ export async function presignedURLCommand(
+
+    if (isNaN(Number(time)) || !Number.isSafeInteger(number) || Number(time) <= 0) {
+        return invalidNumberWarning
+    }
+
+    return undefined
+}
+
+export interface PresignedUrlWizardState {
+    region: string
+    signedUrlParams: SignedUrlRequest
+}
+
+export class PresignedUrlWizard extends Wizard<PresignedUrlWizardState> {
+    constructor(node?: S3FileNode) {
+        // If command is initiated from a S3FileNode, initialize state with region and bucket name
+        // Prompters by default are skipped when the data for their form field is present
+        super({
+            initState: {
+                region: node?.bucket.region,
+                signedUrlParams: node ? ({ bucketName: node.bucket.name } as SignedUrlRequest) : undefined,
+            },
+        })
+        this.form.region.bindPrompter(() => createRegionPrompter().transform(region => region.id))
+        this.form.signedUrlParams.operation.bindPrompter(() => createOperationPrompter())
+        this.form.signedUrlParams.bucketName.bindPrompter(({ region }) =>
+            createBucketPrompter(assertDefined(region, 'region'))
+        )
+
+        if (node) {
+            this.form.signedUrlParams.key.setDefault(node.file.key)
+        }
+
+        this.form.signedUrlParams.key.bindPrompter(
+            ({ region, signedUrlParams }) =>
+                createS3FilePrompter(
+                    assertDefined(region, 'region'),
+                    assertDefined(signedUrlParams?.bucketName, 'bucketName'),
+                    assertDefined(signedUrlParams?.operation, 'operation')
+                ),
+            node ? { showWhen: state => state.signedUrlParams?.operation === 'putObject' } : undefined
+        )
 
-    const time = Number(timeStr)
+        this.form.signedUrlParams.time.bindPrompter(() => createExpiryPrompter().transform(s => Number(s) * 60))
 
-    if (isNaN(time) || time < 0) {
-        throw new ToolkitError('The provided expiration time is not a positive number', { code: 'InvalidInput' })
+        function assertDefined<T>(val: T | undefined, key: string): T {
+            if (val === undefined) {
+                throw Error(`PresignedUrlWizard: "${key}" is undefined`)
+            }
+            return val
+        }
     }
+}
+
+function createOperationPrompter() {
+    const items: DataQuickPickItem<string>[] = [
+        { label: 'Download (GET)', data: 'getObject' },
+        { label: 'Upload (PUT)', data: 'putObject' },
+    ]
 
-    return time
+    return createQuickPick(items, { title: 'Presigned URL: Choose an operation', buttons: createCommonButtons() })
 }
 
-function validateTime(time: string): string | undefined {
-    const number = Number(time)
+function createExpiryPrompter() {
+    return createInputBox({
+        value: '15',
+        prompt: 'Specify the expiry time (minutes) for the URL',
+        placeholder: 'Defaults to 15 minutes',
+        validateInput: validateTime,
+        buttons: createCommonButtons(),
```

Comment:
help button should link to https://docs.aws.amazon.com/AmazonS3/latest/userguide/PresignedUrlUploadObject.html 

later we can update to Toolkit-specific docs.